pingcap/tidb-operator
# TLS support for heterogeneous components

### What problem does this PR solve? 

#3061 
### What is changed and how does it work?
TLS support for heterogeneous components.

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test 
    - 1. Create tls origin cluster and tls heterogeneous cluster which contain one  tiflash 4.0.5 node , they each use independent tls secret.
    - 2. Check that the monitoring and cluster conditions are healthy

Monitor screenshot:
![image](https://user-images.githubusercontent.com/13257127/92637787-edeacf80-f30b-11ea-8957-21320498913e.png)

![image](https://user-images.githubusercontent.com/13257127/92637853-ffcc7280-f30b-11ea-9175-466674e0d464.png)

Cluster info  screenshot:
![image](https://user-images.githubusercontent.com/13257127/92637799-f17e5680-f30b-11ea-993e-7e8c5f1093ca.png)


Code changes

 - Has Go code change
